test/stronger_parameters/controller_support/permitted_parameters_test.rb

```diff
@@ -1,312 +1,274 @@
       )
     end
   end
 
-  describe 'sugar' do
+  describe '.sugar' do
     it 'turns Array into Parameters.array' do
-      WhitelistControllerTester.permitted_parameters :foo, ticket: [Parameters.integer]
-      constraint = WhitelistControllerTester.permitted_parameters_for(:foo)[:ticket]
+      WhitelistsController.permitted_parameters :foo, ticket: [Parameters.integer]
+      constraint = WhitelistsController.permitted_parameters_for(:foo)[:ticket]
       assert_instance_of StrongerParameters::ArrayConstraint, constraint
       assert_equal Parameters.integer, constraint.item_constraint
     end
 
     it 'turns Hash into Parameters.map' do
-      WhitelistControllerTester.permitted_parameters :foo, ticket: { id: Parameters.integer }
-      constraint = WhitelistControllerTester.permitted_parameters_for(:foo)[:ticket]
+      WhitelistsController.permitted_parameters :foo, ticket: { id: Parameters.integer }
+      constraint = WhitelistsController.permitted_parameters_for(:foo)[:ticket]
       assert_instance_of StrongerParameters::HashConstraint, constraint
       assert_equal({ 'id' => Parameters.integer }, constraint.constraints)
     end
   end
 
-  describe 'with prevent_nil_values_in_params' do
-    before do
-      WhitelistControllerTester.permitted_parameters :show, something: {
-        test: {
-          all: Parameters.anything,
-          day: Parameters.anything
-        }
-      }
-      @controller = WhitelistControllerTester.new
-      @controller.request = initialize_test_request('CONTENT_TYPE' => 'application/json')
-      params = { action: 'show', controller: 'test' }.merge(subject)
-      @controller.params = normalize_params(params)
-      @controller.response = initialize_test_request
-      permit_without_raising
+  describe 'emptied containers' do
+    def do_request(params)
+      get :index, params: params, format: :json
     end
 
-    describe 'with child empty hash' do
-      subject { { something: { test: {} } } }
-
-      it 'does not remove test' do
-        assert @controller.params[:something].key?(:test)
-      end
-    end
-
-    describe 'with child empty array' do
-      subject { { something: { test: [] } } }
-
-      it 'does not remove test' do
-        assert @controller.params[:something].key?(:test)
-      end
+    before do
+      WhitelistsController.permitted_parameters :index,
+        something: {
+          test: {
+            all: Parameters.anything,
+            day: Parameters.anything
+          }
+        },
+        array: Parameters.array(Parameters.integer)
     end
 
-    describe 'with empty array' do
-      subject { { something: [] } }
-
-      it 'does not remove something' do
-        assert @controller.params.key?(:something)
-      end
+    # should behave exactly like submitting and empty array does ... but that's a bug in .permit
+    it 'does not removes emptied hash' do
+      do_request something: {remove: 1}
```

Comment:
does it make any difference to test `something: { test: { remove: 1 } }` instead? (guess it'd just nested, that's already being tests somewhere else)